File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/TextLogEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.runner.events.model.error.ExceptionDetail
import com.testerum.runner.events.model.log_level.LogLevel
import com.testerum.runner.events.model.position.EventKey
import java.time.LocalDateTime

data class TextLogEvent @JsonCreator constructor(
        @JsonProperty("time")            override val time: LocalDateTime,
        @JsonProperty("eventKey")        override val eventKey: EventKey,

        @JsonProperty("logLevel")        val logLevel: LogLevel,
        @JsonProperty("message")         val message: String,
        @JsonProperty("exceptionDetail") val exceptionDetail: ExceptionDetail?
): RunnerEvent

File: old/web/web-frontend/src/app/model/test/event/text-log.event.ts
import {RunnerEvent} from "./runner.event";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";
import {ExceptionDetail} from "./fields/exception-detail.model";
import {LogLevel} from "./enums/log-level.enum";

export class TextLogEvent implements RunnerEvent, Serializable<TextLogEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.LOG_EVENT;

    logLevel: LogLevel;
    message: string;
    exceptionDetail: ExceptionDetail | null;

    deserialize(input: Object): TextLogEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.message = input["message"];
        if (input["exceptionDetail"]) {
            this.exceptionDetail = new ExceptionDetail().deserialize(input["exceptionDetail"]);
        }
        this.logLevel = LogLevel[""+input["logLevel"]];


        return this;
    }

    serialize(): string {
        return undefined;
    }

    getMessageWithException(): string {
        let result = "";

        result += this.message;

        if (this.exceptionDetail != null) {
            result += "; exception:\n";
            result += this.exceptionDetail.asDetailedString;
        }

        return result;
    }

}
